java/transforms-by-suitability/src/main/java/com/esri/arcgisruntime/sample/transformsbysuitability/MainActivity.java

```diff
@@ -0,0 +1,342 @@
+    tableList.setOnItemClickListener((AdapterView<?> adapterView, View view, int i, long l) -> {
+
+      view.setSelected(true);
+
+      // Get the datum transformation selected by the user
+      DatumTransformation selectedTransform = (DatumTransformation) adapterView.getAdapter().getItem(i);
+
+      Point projectedGeometry = null;
+      try {
+        // Use the selected transformation to reproject the Geometry
+        projectedGeometry = (Point) GeometryEngine.project(mOriginalGeometry, mMapView.getSpatialReference(),
+            selectedTransform);
+
+      } catch (ArcGISRuntimeException agsEx) {
+        // Catch errors thrown from project method. If a transformation is missing grid files, then it cannot be
+        // successfully used to project a geometry, and will throw an exception.
+        Snackbar.make(tableList, agsEx.getMessage() + "\n" + getResources().getString(R.string.transform_missing_files),
+            Snackbar.LENGTH_LONG).show();
+        removeProjectedGeometryGraphic();
+        return;
+      }
+
+      // Add projected geometry as a second graphic - use a cross symbol which ensures the default transformation
+      // graphic remains visible beneath this graphic.
+      if (projectedGraphic == null) {
+        projectedGraphic = addGraphic(projectedGeometry, Color.argb(255, 255, 0, 0),
+            SimpleMarkerSymbol.Style.CROSS);
+
+      } else {
+        // If graphic already set, just update the geometry
+        projectedGraphic.setGeometry(projectedGeometry);
+      }
+    });
+
+    // If the CheckBox is not checked (default), transformations should be ordered by suitability for the whole
+    // spatial reference. If checked, then transformations will be ordered by suitability for the map extent.
+    CheckBox checkBox = findViewById(R.id.order_by_check_box);
+    checkBox.setOnCheckedChangeListener((CompoundButton compoundButton, boolean newCheckState) -> {
+      // Store the new check state in a member variable and update the list of transformations.
+      mUseExtentForSuitability = newCheckState;
+      setupTransformsList();
+    });
+
+    // Get MapView from layout and set a map into this view
+    mMapView = findViewById(R.id.mapView);
+    mArcGISMap = new ArcGISMap(Basemap.createLightGrayCanvas());
+    mMapView.setMap(mArcGISMap);
+
+    // Create a geometry located in the Greenwich observatory courtyard in London, UK, the location of the
+    // Greenwich prime meridian. This will be projected using the selected transformation.
+    mOriginalGeometry = new Point(538985.355, 177329.516, SpatialReference.create(27700));
+
+    // Add a Graphic to show the original geometry location, projected using the default transformation
+    addGraphic(mOriginalGeometry, Color.argb(255, 0, 0, 255), SimpleMarkerSymbol.Style.SQUARE);
+
+    mArcGISMap.addDoneLoadingListener(() -> {
+      if (mArcGISMap.getLoadStatus() == LoadStatus.LOADED) {
+        if (mTransformValues.size() == 0) {
```

Comment:
This fires for me on device rotation, so the viewpoint is being reset async.

Reply:
This would be the same behaviour in other samples - e.g. offline geocoding. I'm not sure I see a big advantage in adding code to deal with device rotations, unless that's part of the sample workflow, as it would complicate the code.